cub/util_device.cuh

```diff
@@ -538,8 +538,16 @@ CUB_RUNTIME_FUNCTION inline cudaError_t DebugSyncStream(cudaStream_t stream,
   }
 
 #if 1 // All valid targets currently support device-side synchronization
```

Comment:
Should we also remove the conditional compilation and just use the new code unconditionally?

Reply:
Good catch, thanks!

Reply:
Leave this in -- the "dead code" here is a placeholder for an internal usecase.

Reply:
Ignore that -- I missed that the new version still includes both paths 👍